### PR TITLE
Build forever page in store app

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Migrate over the forever page to store app.

## Justify why they are needed

So we can switch 100% to hedvig-com.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2023** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2023" target="_blank">Graphite</a> 
    * **PR #2024** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2024" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2024?utm_source=stack-comment).

File: apps/store/src/pages/forever/[code].tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { GetServerSideProps } from 'next'
+import { isRoutingLocale } from '@/utils/l10n/localeUtils'
+import { PageLink } from '@/utils/PageLink'
+
+const Page = () => {
+  return null
+}
```
Comment (gustaveen) [inline apps/store/src/pages/forever/[code].tsx:7]: Hm, don't really understand what's going on here :D

File: apps/store/src/components/ForeverPage/ForeverPage.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      const [linkText, secondParam] = text.split(']]')
+
+      if (secondParam === undefined) return text
+
+      return (
+        <Link key={linkText} href={PageLink.home()}>
+          {linkText}
+        </Link>
+      )
+    })
```
Comment (gustaveen) [inline apps/store/src/components/ForeverPage/ForeverPage.tsx:35]: Do we need some similar `replaceMarkdown` functionality in other places?

File: apps/store/src/utils/PageLink.ts [hunk 3]
```diff
@@ -52,6 +54,11 @@ export const PageLink = {
   apiPaymentAdyenCallback: ({ locale, shopSessionId }: AdyenCallbackRoute) =>
     `/api/payment/adyen-callback/${shopSessionId}/${locale}`,
 
+  forever: ({ locale, code }: ForeverPage = {}) => {
+    const codeQueryParam = code ? `?code=${code}` : ''
```
Comment (gustaveen) [inline apps/store/src/utils/PageLink.ts:58]: nit: Do we need to rewrite the code as a query param? Looks cleaner if it's not I think :)

**Review by gustaveen — APPROVED** [review]
Great work! 🙌

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/pages/forever/[code].tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/pages/forever/[code].tsx:7]: Ahh this is for backwards compatability -- just wanna redirect to `/se/forever?code=[CODE]`

We actually never render the page

File: apps/store/src/components/CartInventory/useCampaign.ts [hunk 4]
```diff
@@ -3,14 +3,16 @@ import { useErrorMessage } from '@/utils/useErrorMessage'
 
 type Params = {
   cartId: string
+  onCompleted?: () => void
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply apps/store/src/components/CartInventory/useCampaign.ts:6]: Right, no I think we don't -- just expose the things we are actually using

That will make future refactoring easier 😄

File: apps/store/next.config.js [hunk 5]
```diff
@@ -38,6 +38,13 @@ module.exports = withBundleAnalyzer({
       },
     ]
   },
+  rewrites: [
```
Comment (guilhermespopolin) [inline apps/store/next.config.js:41]: Why do we need this again? 🤔

Comment (robinandeer) [reply apps/store/next.config.js:41]: Backwards compatability

It should be possible to use:

- `/se/forever/:code`
- `/se/forever?code=:code`

**Pull request merged**